reconcile/test/test_terraform_vpc_resources.py

```diff
@@ -0,0 +1,127 @@
+import logging
+
+import pytest
+
+from reconcile.gql_definitions.common.app_interface_vault_settings import (
+    AppInterfaceSettingsV1,
+)
+from reconcile.gql_definitions.terraform_vpc_resources.vpc_resources_aws_accounts import (
+    AWSAccountV1,
+    AWSTerraformStateIntegrationsV1,
+    TerraformStateAWSV1,
+    VPCResourcesAWSAccountsQueryData,
+)
+from reconcile.status import ExitCodes
+from reconcile.terraform_vpc_resources import (
+    TerraformVpcResources,
+    TerraformVpcResourcesParams,
+)
+
+
+@pytest.fixture
+def query_data():
+    return VPCResourcesAWSAccountsQueryData(
```

Comment:
`gql_class_factory` might come in handy https://github.com/app-sre/qontract-reconcile/blob/b14feca7affff061773cf6cf712da2de3fec488a/reconcile/test/conftest.py#L124

It helps you creating (nested) data classes and fills the gaps with default values. I.e., you only need to set the values that you are interested in for your test, greatly reducing the boilerplate. 

Example on how this can be used https://github.com/app-sre/qontract-reconcile/blob/b14feca7affff061773cf6cf712da2de3fec488a/reconcile/test/test_terraform_resources.py#L123-L135

Note, that `gql_class_factory` is in our global `conftest.py`, i.e., it is already available for every test you are writing.

Reply:
I'm failing to see how this makes the code better. Seems like you are typing less but dictionaries are harder to read for me than class constructors. Dictionaries also don't have autocomplete.

That said I want to be proved wrong.

Reply:
I think I get now. I needed to add more fields to the query and the more I add test cases more places where I need to update appear. So using `gql_class_factory` and a `something_dict` function can make that easier.

Reply:
But I still could just use fixtures under fixtures.

Reply:
The main advantage is maintainability. `gql_class_factory` adds sane defaults to fields - even if you do not declare them. If you maintain your own fixtures, you will need to adjust them each time sth changes in the data structure.